src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T item, Node<T> next) {
+            this.previous = previous;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addToTail(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Uncorrected index: index out of size");
+        }
+        if (index == size) {
+            addToTail(value);
+        } else {
+            addToHead(value, wantedNode(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkedIndex(index);
+        return wantedNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkedIndex(index);
+        T oldValue = wantedNode(index).item;
+        wantedNode(index).item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkedIndex(index);
+        T removedValue = wantedNode(index).item;
+        erase(wantedNode(index));
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> element = head;
+        for (int i = 0; i < size; i++) {
+            if (element.item == object || (element.item != null && (element.item).equals(object))) {
+                remove(i);
+                return true;
+            }
+            element = element.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> wantedNode(int index) {
```

Comment:
https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names
`Method names are typically verbs or verb phrases. For example, sendMessage or stop.`

Reply:
Chenged